### PR TITLE
Litespeed packages

### LINKED ISSUE #6: Create `lsphp` packages
https://docs.litespeedtech.com/lsws/extapp/php/extensions/

```
apt install lsphpXX-relay
```

### PR DESCRIPTION
**Pull request by tillkruss**
Resolves #6.

- [x] Remove `Provides: php-relay` from Litespeed binaries on Debian
- [x] Compare `lsphp83-relay` and `lsphp83-pecl-redis` control data (DEB)
- [x] Compare `lsphp83-relay` and `lsphp83-pecl-redis` control data (RPM)
- [x] Add all `ls` package definitions to `fpm.sh`

### CODE REVIEW

**Comment by tillkruss** [comment]
@yatsukhnenko Can you build the packages locally and see 1) if they install with lsphp and 2) post a comparison of the package metadata redis vs relay?

With metadata I mean the control data that lists all the `depends`, `provides` and other fields.

**Comment by yatsukhnenko** [comment]
| redis                                                                     | relay                                                                                                                    |
|---------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
|dpkg-deb -I /var/www/vhosts/lsphp83-redis_6.0.2-3+noble_amd64.deb          | dpkg -I dist/lsphp83-relay-0.9.0-php8.3-debian_arm64.deb                                                                 |
|new Debian package, version 2.0.                                           | new Debian package, version 2.0.                                                                                         |
| size 204034 bytes: control archive=628 bytes.                             |  size 1130050 bytes: control archive=743 bytes.                                                                          |
|                                                                           |      64 bytes,     1 lines      conffiles                                                                                |
|     463 bytes,    13 lines      control                                   |      425 bytes,    13 lines      control                                                                                 |
|     334 bytes,     4 lines      md5sums                                   |      255 bytes,     3 lines      md5sums                                                                                 |
|                                                                           |      185 bytes,     9 lines   *  postinst             #!/bin/bash                                                        |
| Package: lsphp83-redis                                                    | Package: lsphp83-relay                                                                                                   |
| Version: 6.0.2-3+noble                                                    | Version: 0.9.0                                                                                                           |
|                                                                           | License: Proprietary                                                                                                     |
|                                                                           | Vendor: CacheWerk, Inc.                                                                                                  |
| Architecture: amd64                                                       | Architecture: arm64                                                                                                      |
| Maintainer: lsadm <info@litespeedtech.com>                                | Maintainer: Relay Team <hello@cachewerk.com>                                                                             |
| Installed-Size: 815                                                       | Installed-Size: 3470                                                                                                     |
| Depends: lsphp83-igbinary, lsphp83-common, libc6 (>= 2.38)                | Depends: libc6 (>= 2.17), liblz4-1 (>= 0.0~r130), libzstd1 (>= 1.3.2), lsphp83-common, lsphp83-igbinary, lsphp83-msgpack |
|                                                                           | Provides: php-relay                                                                                                      |
| Suggests: redis-server                                                    |                                                                                                                          |
| Section: php                                                              | Section: php                                                                                                             |
| Priority: optional                                                        | Priority: optional                                                                                                       |
|                                                                           | Homepage: https://relay.so                                                                                               |
| Description: PHP extension for interfacing with Redis                     | Description: The next-generation caching layer for PHP.                                                                  |
|  This extension allows php applications to communicate with the Redis     |                                                                                                                          |
|  persistent key-value store.  The php-redis module provides an easy       |                                                                                                                          |
|  object oriented interface.                                               |                                                                                                                          |

**Comment by yatsukhnenko** [comment]
| redis                                                                                                          | relay                                                    |
|----------------------------------------------------------------------------------------------------------------|----------------------------------------------------------|
| rpm -qi var/cache/dnf/litespeed-update-cbc5271992c3ccdb/packages/lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64.rpm | rpm -qi dist/lsphp83-relay-0.9.0-php8.3-el9_x86_64.rpm   |
| Name        : lsphp83-pecl-redis                                                                               | Name        : lsphp83-relay                              |
| Version     : 6.0.2                                                                                            | Version     : 0.9.0                                      |
| Release     : 3.el8.8.3                                                                                        | Release     : 1                                          |
| Architecture: x86_64                                                                                           | Architecture: x86_64                                     |
| Install Date: (not installed)                                                                                  | Install Date: (not installed)                            |
| Group       : Development/Languages                                                                            | Group       : php                                        |
| Size        : 1321629                                                                                          | Size        : 6945998                                    |
| License     : PHP                                                                                              | License     : Proprietary                                |
| Signature   : DSA/SHA256, Tue Jul 30 01:14:27 2024, Key ID 3f6f627083084d0e                                    | Signature   : (none)                                     |
| Source RPM  : lsphp83-pecl-redis-6.0.2-3.el8.8.3.src.rpm                                                       | Source RPM  : lsphp83-relay-0.9.0-1.src.rpm              |
| Build Date  : Tue Jul 30 00:45:26 2024                                                                         | Build Date  : Tue Nov 26 13:25:34 2024                   |
| Build Host  : f38-build                                                                                        | Build Host  : a79f969cc9f7                               |
| Relocations : (not relocatable)                                                                                | Relocations : /                                          |
|                                                                                                                | Packager    : Relay Team <hello@cachewerk.com>           |
|                                                                                                                | Vendor      : CacheWerk, Inc.                            |
| URL         : http://pecl.php.net/package/redis                                                                | URL         : https://relay.so                           |
| Summary     : Extension for communicating with the Redis key-value store                                       | Summary     : The next-generation caching layer for PHP. |
| Description :                                                                                                  | Description :                                            |
| The phpredis extension provides an API for communicating                                                       | The next-generation caching layer for PHP.               |
| with the Redis key-value store.                                                                                |                                                          |
|                                                                                                                |                                                          |
| This Redis client implements most of the latest Redis API.                                                     |                                                          |
| As method only only works when also implemented on the server side,                                            |                                                          |
| some doesn't work with an old redis server version.                                                            |                                                          |
|                                                                                                                |                                                          |
| Package built for PHP 8.3.                                                                                     |                                                          |

**Comment by tillkruss** [comment]
We gotta remove “Provides: php-relay”

**Comment by tillkruss** [comment]
@yatsukhnenko can you post the dependencies for both rpm packages? Probably a dedicated command.

**Comment by yatsukhnenko** [comment]
**Redis**
```
rpm -qR var/cache/dnf/litespeed-update-cbc5271992c3ccdb/packages/lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64.rpm
/bin/sh
/bin/sh
/bin/sh
config(lsphp83-pecl-redis) = 6.0.2-3.el8.8.3
libc.so.6()(64bit)
libc.so.6(GLIBC_2.14)(64bit)
libc.so.6(GLIBC_2.2.5)(64bit)
libc.so.6(GLIBC_2.3)(64bit)
libc.so.6(GLIBC_2.3.4)(64bit)
libc.so.6(GLIBC_2.4)(64bit)
liblzf.so.1()(64bit)
lsphp83(api) = 20230831
lsphp83(zend-abi) = 20230831
lsphp83-json
lsphp83-pecl-igbinary(x86-64)
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
rtld(GNU_HASH)

rpm -i var/cache/dnf/litespeed-update-cbc5271992c3ccdb/packages/lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64.rpm
error: Failed dependencies:
	liblzf.so.1()(64bit) is needed by lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64
	lsphp83(api) = 20230831 is needed by lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64
	lsphp83(zend-abi) = 20230831 is needed by lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64
	lsphp83-json is needed by lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64
	lsphp83-pecl-igbinary(x86-64) is needed by lsphp83-pecl-redis-6.0.2-3.el8.8.3.x86_64
```

**Relay**
```
rpm -qR lsphp83-relay-0.9.0-php8.3-el8_x86_64.rpm
/bin/sh
libzstd
lsphp83(api) = 20230831
lsphp83-pecl-igbinary
lsphp83-pecl-msgpack
lz4
openssl
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1

rpm -i lsphp83-relay-0.9.0-php8.3-el8_x86_64.rpm
error: Failed dependencies:
	lsphp83(api) = 20230831 is needed by lsphp83-relay-0.9.0-1.x86_64
	lsphp83-pecl-igbinary is needed by lsphp83-relay-0.9.0-1.x86_64
	lsphp83-pecl-msgpack is needed by lsphp83-relay-0.9.0-1.x86_64
	lz4 is needed by lsphp83-relay-0.9.0-1.x86_64
	openssl is needed by lsphp83-relay-0.9.0-1.x86_64
```

**Comment by tillkruss** [comment]
@yatsukhnenko I feel like this looks good, theirs is stricter and more extensive but I think it will work anyhow.

Can you figure out why our package has `Provides: php-relay` even tho `pkg_provides` isn't set in our config?

Maybe we should provide it as `lsphp-relay`, if it can't be removed?

**Comment by yatsukhnenko** [comment]
@tillkruss It looks like we set value in `config.multi.sh` and it apply to all next builds
https://github.com/cachewerk/linux-repos/blob/main/build/fpm/src/deb/config.multi.sh#L4

We should set empty `pkg_provides` in `config.ls.sh` or build `ls` targets before `multi`

File: build/fpm/src/deb/config.ls.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+pkg_name="lsphp$php_version_short-relay"
+pkg_identifier="debian"
+
```
Comment (yatsukhnenko) [inline build/fpm/src/deb/config.ls.sh:5]:
```suggestion
pkg_provides=""

```

Comment (yatsukhnenko) [reply build/fpm/src/deb/config.ls.sh:5]: With change above there is no `Provides` field
```
 dpkg -I dist/lsphp74-relay-0.9.0-php7.4-debian_amd64.deb
 new Debian package, version 2.0.
 size 992246 bytes: control archive=735 bytes.
      64 bytes,     1 lines      conffiles            
     405 bytes,    12 lines      control              
     255 bytes,     3 lines      md5sums              
     185 bytes,     9 lines   *  postinst             #!/bin/bash
 Package: lsphp74-relay
 Version: 0.9.0
 License: Proprietary
 Vendor: CacheWerk, Inc.
 Architecture: amd64
 Maintainer: Relay Team <hello@cachewerk.com>
 Installed-Size: 2846
 Depends: libc6 (>= 2.17), liblz4-1 (>= 0.0~r130), libzstd1 (>= 1.3.2), lsphp74-common, lsphp74-igbinary, lsphp74-msgpack
 Section: php
 Priority: optional
 Homepage: https://relay.so
 Description: The next-generation caching layer for PHP.
```

**Comment by tillkruss** [comment]
@yatsukhnenko oh dang. Should we maybe just unset all variables inside our helper file so it's generic?

**Comment by yatsukhnenko** [comment]
@tillkruss something like this?
```diff
diff --git a/build/fpm/helpers.sh b/build/fpm/helpers.sh
index 8d73efd..8296d2c 100755
--- a/build/fpm/helpers.sh
+++ b/build/fpm/helpers.sh
@@ -7,6 +7,7 @@ main()
   echo -n "$version" > /root/fpm/dist/TAG
 
   for package in "${packages[@]}"; do
+    unset ${!pkg_@}
     fpm_build $package
   done
```

**Comment by tillkruss** [comment]
Yeah if that works?

**Comment by yatsukhnenko** [comment]
It might be `unset ${pkg_*}`
I need to test both

**Comment by yatsukhnenko** [comment]
@tillkruss https://github.com/cachewerk/linux-repos/pull/20#issuecomment-2504671617 works :ok:

**Comment by tillkruss** [comment]
Let’ do it.

**Pull request merged**